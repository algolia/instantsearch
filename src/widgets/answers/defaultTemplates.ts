import { AnswersComponentTemplates } from '../../components/Answers/Answers';

const defaultTemplates: AnswersComponentTemplates = {
  header: '',
  loader: '',
  item: (item) => JSON.stringify(item),
};

export default defaultTemplates;
