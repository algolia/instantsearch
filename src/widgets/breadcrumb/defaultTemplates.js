export default {
  home: 'Home',
  separator: ' > ',
};
