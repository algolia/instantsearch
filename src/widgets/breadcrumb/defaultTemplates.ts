import type { BreadcrumbComponentTemplates } from '../../components/Breadcrumb/Breadcrumb';

const defaultTemplates: BreadcrumbComponentTemplates = {
  home: 'Home',
  separator: '>',
};

export default defaultTemplates;
