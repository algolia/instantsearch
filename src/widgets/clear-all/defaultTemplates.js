export default {
  header: '',
  link: 'Clear all',
  footer: ''
};
