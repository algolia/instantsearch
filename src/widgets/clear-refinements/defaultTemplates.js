export default {
  button: 'Clear all',
};
