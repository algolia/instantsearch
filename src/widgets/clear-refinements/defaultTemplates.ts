import type { ClearRefinementsComponentTemplates } from '../../components/ClearRefinements/ClearRefinements';

const defaultTemplates: ClearRefinementsComponentTemplates = {
  resetLabel() {
    return 'Clear refinements';
  },
};

export default defaultTemplates;
