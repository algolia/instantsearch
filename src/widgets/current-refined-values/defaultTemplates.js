export default {
  header: '',
  item: '' +
    '{{#label}}' +
      '{{label}}' +
      '{{^operator}}:{{/operator}}' +
      ' ' +
    '{{/label}}' +
    '{{#operator}}{{{displayOperator}}} {{/operator}}' +
    '{{#exclude}}-{{/exclude}}' +
    '{{name}} <span class="{{cssClasses.count}}">{{count}}</span>',
  clearAll: 'Clear all',
  footer: ''
};
