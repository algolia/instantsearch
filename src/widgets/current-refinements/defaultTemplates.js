export default {
  item: '<span class="{{cssClasses.categoryLabel}}">{{computedLabel}}</span>',
};
