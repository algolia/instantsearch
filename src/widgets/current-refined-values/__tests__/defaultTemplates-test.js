import defaultTemplates from '../defaultTemplates.js';

describe('current-refined-values defaultTemplates', () => {
  describe('`item` template', () => {
    it('has a `item` default template', () => {
      const item = {
        type: 'disjunction',
        label: 'Brand',
        operator: ':',
        name: 'Samsung',
        count: 4,
        cssClasses: {
          count: 'ais-current-refined-values--count',
        },
      };
      expect(defaultTemplates.item(item)).toContain(
        '<span class="ais-current-refined-values--count">4</span>'
      );
      expect(defaultTemplates.item(item)).toMatchSnapshot();
    });
    it('wraps query refinements with <q>', () => {
      const item = {
        type: 'query',
        label: 'Query',
        operator: ':',
        name: 'Samsu',
      };
      expect(defaultTemplates.item(item)).toContain('Query :   <q>Samsu</q>');
      expect(defaultTemplates.item(item)).toMatchSnapshot();
    });
    it('does not show `count` when query refinement', () => {
      const item = {
        type: 'query',
        label: 'Query',
        operator: ':',
        name: 'Samsu',
        count: 22,
      };
      expect(defaultTemplates.item(item)).not.toContain(22);
      expect(defaultTemplates.item(item)).toMatchSnapshot();
    });
  });
});
