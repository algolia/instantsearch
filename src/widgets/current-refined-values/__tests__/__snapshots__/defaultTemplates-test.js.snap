// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`current-refined-values defaultTemplates \`item\` template does not show \`count\` when query refinement 1`] = `"Query :   <q>Samsu</q> "`;

exports[`current-refined-values defaultTemplates \`item\` template has a \`item\` default template 1`] = `"Brand :   Samsung <span class=\\"ais-current-refined-values--count\\">4</span>"`;

exports[`current-refined-values defaultTemplates \`item\` template wraps query refinements with <q> 1`] = `"Query :   <q>Samsu</q> "`;
